client/components/UploadDropzone/UploadDropzone.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import Dropzone from 'react-dropzone'
+import CloudUploadIcon from 'material-ui/svg-icons/file/cloud-upload'
+import Snackbar from 'material-ui/Snackbar'
+
+import styles from './UploadDropzone.scss'
+
+const defaultTypes = ['image/jpeg', 'image/png', 'image/bmp']
+
+class UploadDropzone extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      file: this.props.file || null,
+      sizeLimit: this.props.sizeLimit || 3000000,
+      types: this.props.types || defaultTypes,
+      error: false,
+      errorMessage: ''
+    }
+  }
+  isImage (file) {
+    const name = file.name || file.path
+    const suffix = name.split('.').reverse()[0].toLowerCase()
+
+    return suffix === 'jpg' || suffix === 'jpeg' || suffix === 'bmp' || suffix === 'png'
```

Comment:
As you have defaultTypes, you could have an array of validImageExtensions and then make validImageExtensions.indexOf(suffix) !== -1

Reply:
also to get the file extension...
name.split('.').pop(); :)

Reply:
I think that it can be done even more simply.. though because react-dropzone already verifies that the uploaded file is an image before it actually gets uploaded, it seems redundant to have an `isImage` function in the first place. I just wasn't sure if this component may in the future be reused for non-image uploading, so I didn't want to leave out the possibility of that

though perhaps its best to leave that function out until we actually need it? what do you think?

Reply:
I think we could have a folder like "utils" or so, that could contain this kind of validations. I think that in that Utils should be the SignupValidator (the name could be more generic, like, Validators), and something like just Utils that could contain methods like isImage, convertTo (a date for example to a specific mask), etc, etc.